src/pages/TokenDetails/index.tsx

```diff
@@ -14,38 +14,41 @@ export const pageTimePeriodAtom = atomWithStorage<TimePeriod>('tokenDetailsTimeP
 export default function TokenDetailsPage() {
   const { tokenAddress, chainName } = useParams<{ tokenAddress?: string; chainName?: string }>()
   const chain = validateUrlChainParam(chainName)
-  const pageChainId = CHAIN_NAME_TO_CHAIN_ID[chain]
   const isNative = tokenAddress === NATIVE_CHAIN_ID
   const [timePeriod, setTimePeriod] = useAtom(pageTimePeriodAtom)
-  const [contract, duration] = useMemo(
-    () => [
-      { address: isNative ? nativeOnChain(pageChainId).wrapped.address : tokenAddress ?? '', chain },
-      toHistoryDuration(timePeriod),
-    ],
-    [chain, isNative, pageChainId, timePeriod, tokenAddress]
+  const [address, duration] = useMemo(
+    () => [isNative ? getNativeTokenDBAddress(chain) : tokenAddress ?? '', toHistoryDuration(timePeriod)],
```

Comment:
if you set up the `tokenAddress === NATIVE_CHAIN_ID` check as a typescript guard function like:

```isNativeAddress(tokenAddress: string): tokenAddress is string { return  tokenAddress === NATIVE_CHAIN_ID}
```

then you'd be able to skip this `tokenAddress ?? ''` coalescing bcuz typescript will know it's always defined :P

Reply:
but then a non-native address would be incorrectly seen by typescript as NOT a string?